build-system/compile/post-closure-babel.js

```diff
@@ -86,22 +86,23 @@ exports.postClosureBabel = function () {
       file.contents,
       file.sourceMap
     );
-    const {compressed, terserMap} = terserMinify(code);
 
-    file.contents = Buffer.from(compressed, 'utf-8');
-    file.sourceMap = remapping(
-      [terserMap, babelMap, map],
-      () => null,
-      !argv.full_sourcemaps
-    );
+    terserMinify(code).then(({compressed, terserMap}) => {
```

Comment:
We've been replacing all `then()` chains in `build-system` with `async/await` (#26562). Would  something like this  work instead?

```js
const {compressed, terserMap} = await terserMinify(code);
file.contents = Buffer.from(...
```

Reply:
Unfortunately it appears that `through` requires these methods to not return Promises.

```typescript
declare namespace through2 {
    interface Through2Constructor extends stream.Transform {
        new (opts?: stream.DuplexOptions): stream.Transform;
        (opts?: stream.DuplexOptions): stream.Transform;
    }

    type TransformCallback = (err?: any, data?: any) => void;
    type TransformFunction = (this: stream.Transform, chunk: any, enc: BufferEncoding, callback: TransformCallback) => void;
    type FlushCallback = (this: stream.Transform, flushCallback: () => void) => void;

    /**
     * Convenvience method for creating object streams
     */
    function obj(transform?: TransformFunction, flush?: FlushCallback): stream.Transform;

    /**
     * Creates a constructor for a custom Transform. This is useful when you
     * want to use the same transform logic in multiple instances.
     */
    function ctor(transform?: TransformFunction, flush?: FlushCallback): Through2Constructor;
    function ctor(opts?: stream.DuplexOptions, transform?: TransformFunction, flush?: FlushCallback): Through2Constructor;
}
```

Reply:
Although I do see a few places where it is using an `async` method in our codebase. I'll take a look.

Reply:
Turns out I was misreading this! I am testing the change to use `async/await` now. Thanks for the suggestion!